src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+        System.out.println("\t   unmark [x]                                     Unmark task number x in your list");
+        System.out.println("\t   delete [x]                                     Delete task number x in your list");
+        System.out.println("\t   todo [task name]                               Add a ToDo to your list");
+        System.out.println("\t   deadline [task name] /by [deadline]            Add a Deadline to your list");
+        System.out.println("\t   event [task name] /from [start] /to [end]      Add an Event to your list");
+        System.out.println("\t____________________________________________________________\n");
+    }
+
+    // Task list mode, allows user to enter tasks into a list, and print all tasks in the list
+    // currently with "list", exits with "bye"
+    private static void taskListMode() {
+        Scanner scanner = new Scanner(System.in);
+        final Pattern markPattern = Pattern.compile("^mark\\s(\\d+)$");
+        final Pattern unmarkPattern = Pattern.compile("^unmark\\s(\\d+)$");
+        final Pattern deletePattern = Pattern.compile("^delete\\s(\\d+)$");
+        final Pattern todoPattern = Pattern.compile("^todo\\s(.+)$");
+        final Pattern deadlinePattern = Pattern.compile("^deadline\\s(.+)\\s/by\\s(.+)$");
+        final Pattern eventPattern = Pattern.compile("^event\\s(.+)\\s/from\\s(.+)\\s/to\\s(.+)$");
+        // Execution loop
+        while (true) {
+            // Command line input, stripped of whitespaces
+            String inputText = scanner.nextLine().strip();
+            if (inputText.equals("bye")) {
+                // Exit task list mode
+                break;
+            } else if (inputText.equals("help")) {
+                // List all available commands
+                help();
+            } else if (inputText.equals("list")) {
+                // List all tasks in taskList currently, or return a message indicating that there are none
+                int taskListSize = taskList.size();
+                System.out.println("\t____________________________________________________________");
+                if (taskListSize == 0) {
+                    // List is empty
+                    System.out.println("\t Looks like there aren't any tasks in your list yet... let's add some!");
+                } else {
+                    // List has items
+                    System.out.println("\t Let's take a look at the tasks in your list:");
+                    // Print list of tasks
+                    for (int i = 0; i < taskListSize; i++) {
+                        Task task = taskList.get(i);
+                        System.out.println("\t " + (i + 1) + "." + task.detailsAsString());
+                    }
+                }
+                System.out.println("\t____________________________________________________________\n");
+            } else if (inputText.startsWith("mark")) {
```

Comment:
Cosider Having functions inside the else if block which deals with the logic instead of having a big block of logic. That will make the code more readable. EG:

```suggestion
            } else if (inputText.startsWith("mark")) {
                handleMark(inputText);
            }
```